config/database_test.go

```diff
@@ -467,12 +466,31 @@ func TestDatabaseStoreSet(t *testing.T) {
 
 		_, err = ds.Set(newCfg)
 		if assert.Error(t, err) {
-			assert.True(t, strings.HasPrefix(err.Error(), "failed to persist: failed to write to database"))
+			t.Log(err.Error())
+			assert.True(t, strings.HasPrefix(err.Error(), "failed to persist: failed to query active configuration"))
 		}
 
 		assert.Equal(t, "", *ds.Get().ServiceSettings.SiteURL)
 	})
 
+	t.Run("persist failed: too long", func(t *testing.T) {
+		_, tearDown := setupConfigDatabase(t, emptyConfig, nil)
+		defer tearDown()
+
+		ds, err := config.NewDatabaseStore(fmt.Sprintf("%s://%s", *sqlSettings.DriverName, *sqlSettings.DataSource))
+		require.NoError(t, err)
+		defer ds.Close()
+
+		longSiteURL := fmt.Sprintf("http://%s", strings.Repeat("a", config.MaxWriteLength))
+		newCfg := emptyConfig.Clone()
+		newCfg.ServiceSettings.SiteURL = sToP(longSiteURL)
+
+		_, err = ds.Set(newCfg)
+		if assert.Error(t, err) {
```

Comment:
Why not `require.Error(t, err)` and then `assert.True`? I think is more plain and more readable, and there is not reason to use the if, because you don't have any code after that.